device/tt_device.h

```diff
@@ -23,6 +23,9 @@
 #include "pcie/pci_device.hpp"
 #include "fmt/core.h"
 
+constexpr std::uint32_t NOC_ADDR_LOCAL_BITS = 36; // source: noc_parameters.h, common for WH && BH
```

Comment:
Why are we not defining these constants in a separate file, same as ethernet constants. Then we could have same things in constructor for NOC params

```cpp
noc_params = architecture_implementation->get_noc_params();
```

It is not that I am not ok with this change at the moment, but in the future we will surely move this around and if it is separated into a file then we could keep it around while redesigning everything but keep param logic there. Let me know if this makes sense

Reply:
I took path of least resistance. What you are saying makes sense though.

Reply:
I just did a quick refactor.